### PR TITLE
Two new modules: agat convertspgxf2gxf and convertspgff2gtf. Issue  #2527

### PR DESCRIPTION
**Pull request by toniher**
This basically adds two tools for conversion of GFF and GTF formats

### CODE REVIEW

**Comment by toniher** [comment]
@nf-core-bot fix linting

File: modules/nf-core/agat/convertspgxf2gxf/main.nf [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    """
+    agat_convert_sp_gxf2gxf.pl \\
+        --gff $input \\
+        --output ${prefix}.agat.gff \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        agat: \$(agat_convert_sp_gxf2gxf.pl --help |head -n4 | tail -n1 | grep -Eo '[0-9.]+')
```
Comment (mahesh-panchal) [inline modules/nf-core/agat/convertspgxf2gxf/main.nf:33]:
```suggestion
        agat: \$(agat_convert_sp_gxf2gxf.pl --help | sed '4!d; s/.*v//')
```

File: modules/nf-core/agat/convertspgff2gtf/main.nf [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    """
+    agat_convert_sp_gff2gtf.pl \\
+        --gff $input \\
+        --output ${prefix}.agat.gtf \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        agat: \$(agat_convert_sp_gff2gtf.pl --help |head -n4 | tail -n1 | grep -Eo '[0-9.]+')
```
Comment (mahesh-panchal) [inline modules/nf-core/agat/convertspgff2gtf/main.nf:33]:
```suggestion
        agat: \$(agat_convert_sp_gff2gtf.pl --help | sed '4!d; s/.*v//')
```

File: modules/nf-core/agat/convertspgff2gtf/main.nf [hunk 3]
```diff
@@ -0,0 +1,36 @@
+process AGAT_CONVERTSPGFF2GTF {
+    tag "$meta.id"
+    label 'process_low'
```
Comment (mahesh-panchal) [inline modules/nf-core/agat/convertspgff2gtf/main.nf:3]:
```suggestion
    label 'process_single'
```

File: modules/nf-core/agat/convertspgxf2gxf/main.nf [hunk 4]
```diff
@@ -0,0 +1,36 @@
+process AGAT_CONVERTSPGXF2GXF {
+    tag "$meta.id"
+    label 'process_low'
```
Comment (mahesh-panchal) [inline modules/nf-core/agat/convertspgxf2gxf/main.nf:3]:
```suggestion
    label 'process_single'
```

File: modules/nf-core/agat/convertspgff2gtf/main.nf [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    conda (params.enable_conda ? "bioconda::agat=1.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/agat:1.0.0--pl5321hdfd78af_0' :
+        'quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(input)
```
Comment (mahesh-panchal) [inline modules/nf-core/agat/convertspgff2gtf/main.nf:11]:
```suggestion
    tuple val(meta), path(gff)
```
Use `gff` to be consistent with your meta.yml

File: modules/nf-core/agat/convertspgxf2gxf/main.nf [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+    conda (params.enable_conda ? "bioconda::agat=1.0.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/agat:1.0.0--pl5321hdfd78af_0' :
+        'quay.io/biocontainers/agat:1.0.0--pl5321hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(input)
```
Comment (mahesh-panchal) [inline modules/nf-core/agat/convertspgxf2gxf/main.nf:11]:
```suggestion
    tuple val(meta), path(gff)
```
Use `gff` to be consistent with the meta.yml

File: tests/modules/nf-core/agat/convertspgff2gtf/main.nf [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+nextflow.enable.dsl = 2
+
+include { AGAT_CONVERTSPGFF2GTF } from '../../../../../modules/nf-core/agat/convertspgff2gtf/main.nf'
+
+workflow test_gff {
+
+    input = [ file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true) ]
+
+    AGAT_CONVERTSPGFF2GTF ( [ [id:'test'], input ] )
```
Comment (mahesh-panchal) [inline tests/modules/nf-core/agat/convertspgff2gtf/main.nf:11]:
```suggestion
    input = [ 
        [ id: 'test' ], // meta map
        file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true) 
    ]

    AGAT_CONVERTSPGFF2GTF ( input )
```
Please use the same format as the other modules.

File: tests/modules/nf-core/agat/convertspgxf2gxf/main.nf [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+nextflow.enable.dsl = 2
+
+include { AGAT_CONVERTSPGXF2GXF } from '../../../../../modules/nf-core/agat/convertspgxf2gxf/main.nf'
+
+workflow test_gff {
+
+    input = [ file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true) ]
+
+    AGAT_CONVERTSPGXF2GXF ( [ [id:'test'], input ] )
+}
```
Comment (mahesh-panchal) [inline tests/modules/nf-core/agat/convertspgxf2gxf/main.nf:12]:
```suggestion
    input = [
        [ id: 'test' ], // meta map
        file(params.test_data['sarscov2']['genome']['genome_gff3'], checkIfExists: true) 
    ]

    AGAT_CONVERTSPGXF2GXF ( input )
}
```
Please use the same format as other modules.

File: tests/modules/nf-core/agat/convertspgxf2gxf/main.nf [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+
+workflow test_gtf {
+
+    input = [ file(params.test_data['sarscov2']['genome']['genome_gtf'], checkIfExists: true) ]
+
+    AGAT_CONVERTSPGXF2GXF ( [ [id:'test'], input ] )
```
Comment (mahesh-panchal) [inline tests/modules/nf-core/agat/convertspgxf2gxf/main.nf:18]:
```suggestion
    input = [
        [ id: 'test' ], // meta map
        file(params.test_data['sarscov2']['genome']['genome_gtf'], checkIfExists: true) 
    ]

    AGAT_CONVERTSPGXF2GXF ( input )
```
Same format as other modules.

**Review by mahesh-panchal — COMMENTED** [review]
Almost there. Thank you for adding these.

**Review by mahesh-panchal — APPROVED** [review]
Nice work.

**Pull request merged**